iothub_client/inc/iothub_device_client.h

```diff
@@ -349,6 +348,133 @@ extern "C"
+    *           This function is expected to be used along with:
+    *           `IoTHubDeviceClient_CreateUploadContext`
+    *           `IoTHubDeviceClient_AzureStoragePutBlock`
+    *           `IoTHubDeviceClient_DestroyUploadContext`
+    *           For the standard less-granular uploads to blob please use either
+    *           `IoTHubDeviceClient_UploadToBlobAsync` or `IoTHubDeviceClient_UploadMultipleBlocksToBlobAsync`.
+    *
+    * @param    iotHubClientHandle      The handle created by a call to the create function.
+    * @param    azureBlobSasUri         The Azure Storage Blob SAS uri obtained with `IoTHubDeviceClient_InitializeUpload`.
+    *
+    * @warning  This is a synchronous/blocking function.
+    *
+    * @return   A `IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE` on success or NULL if the function fails.
+    */
+    MOCKABLE_FUNCTION(, IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE, IoTHubDeviceClient_CreateUploadContext, IOTHUB_DEVICE_CLIENT_HANDLE, iotHubClientHandle, const char*, azureBlobSasUri);
+
+    /**
+    * @brief    This API upload a single blob block to Azure Storage (performs a PUT BLOCK operation).
+    * @remark   It is part of a set of functions for more granular control over Azure IoT-based blob uploads.
+    *           This function is expected to be used along with:
+    *           `IoTHubDeviceClient_CreateUploadContext`
+    *           `IoTHubDeviceClient_AzureStoragePutBlock`
+    *           `IoTHubDeviceClient_DestroyUploadContext`
+    *           For the standard less-granular uploads to blob please use either
+    *           `IoTHubDeviceClient_UploadToBlobAsync` or `IoTHubDeviceClient_UploadMultipleBlocksToBlobAsync`.
+    *
+    * @param    uploadContextHandle    The handle created with `IoTHubDeviceClient_CreateUploadContext`.
+    * @param    blockNumber            Number of the block being uploaded.
+    * @param    dataPtr                Pointer to the block data to be uploaded to Azure Storage blob.
+    * @param    dataSize               Size of the block data pointed by `dataPtr`.
+    *
+    * @warning  This is a synchronous/blocking function.
+    *
+    * @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
+    */
+    MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubDeviceClient_AzureStoragePutBlock, IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE, uploadContextHandle, uint32_t, blockNumber, const uint8_t*, dataPtr, size_t, dataSize);
+
+    /**
+    * @brief    This API performs an Azure Storage PUT BLOCK LIST operation.
+    * @remark   It is part of a set of functions for more granular control over Azure IoT-based blob uploads.
+    *           This function is expected to be used along with:
+    *           `IoTHubDeviceClient_CreateUploadContext`
+    *           `IoTHubDeviceClient_AzureStoragePutBlock`
+    *           `IoTHubDeviceClient_DestroyUploadContext`
```

Comment:
```suggestion
    *           `IoTHubDeviceClient_InitializeUpload`
    *           `IoTHubDeviceClient_CreateUploadContext`
    *           `IoTHubDeviceClient_DestroyUploadContext`
```

not sure about this one

Reply:
Fixed.